packages/server/src/service-manager.ts

```diff
@@ -6,8 +6,10 @@ import { PageConfig, URLExt } from '@jupyterlab/coreutils';
 import { IServiceWorkerManager, WORKER_NAME } from './tokens';
 
 export class ServiceWorkerManager implements IServiceWorkerManager {
-  constructor() {
-    void this.initialize().catch(console.warn);
+  constructor(options?: IServiceWorkerManager.IOptions) {
+    const workerUrl =
+      options?.workerUrl ?? URLExt.join(PageConfig.getBaseUrl(), WORKER_NAME);
```

Comment:
@trungleduc @jtpio is there a way to set this option? from what I can see here: https://github.com/jupyterlite/jupyterlite/blob/bde12e5209c37e278c5b84af9ebe40fc17696475/packages/server/src/app.ts#L65-L66

there's no way to pass this in when creating a new instance of the JupyterLiteServer, and no way to load options onto the page

Reply:
you can disable the `"@jupyterlite/server-extension:service-worker"` plugin and provide your own `IServiceWorkerManager`
https://github.com/jupyterlite/jupyterlite/blob/bde12e5209c37e278c5b84af9ebe40fc17696475/packages/server-extension/src/index.ts#L200-L207

Reply:
cool - I take it that you mean adding my own plugin with this `id: '@jupyterlite/server-extension:service-worker'` will automatically override / disable the default plugin, and of course replace it with the new one?